src/electrum_wallet_file.rs

```diff
@@ -0,0 +1,438 @@
+use crate::{ElectrumExtendedKey, ElectrumExtendedPrivKey, ElectrumExtendedPubKey};
+use bitcoin::util::bip32::{ExtendedPrivKey, ExtendedPubKey};
+use regex::Regex;
+use serde::{de, ser::SerializeMap, Deserialize, Deserializer, Serialize, Serializer};
+use std::{fmt, path::Path, str::FromStr, string::ToString};
+
+/// Representation of an electrum wallet file
+#[derive(Clone, Debug, PartialEq)]
+pub struct ElectrumWalletFile {
+    pub addresses: Addresses,
+    pub wallet_type: WalletType,
+    pub keystores: Vec<Keystore>,
+}
+
+impl ElectrumWalletFile {
+    /// Parse an electrum wallet file
+    pub fn from_file(wallet_file: &Path) -> Result<Self, String> {
+        let file = std::fs::File::open(wallet_file).map_err(|e| e.to_string())?;
+        let wallet = serde_json::from_reader(file).map_err(|e| e.to_string())?;
```

Comment:
Oh, that's something clippy should report then.

Reply:
IIUC I think it would not be easy for clippy to discriminate if the file is read byte by byte (needs BufReader) or in a whole (doesn't need BufReader)